packages/flutter_tools/lib/src/test/test_compiler.dart

```diff
@@ -179,20 +199,4 @@ class TestCompiler {
       compilationQueue.removeAt(0);
     }
   }
-
-  void _reportCompilerMessage(String message, {bool emphasis, TerminalColor color}) {
```

Comment:
Rather than printing this message after a bunch of compilation failures, the new test compiler checks for a dependency before compiling and exits early. This gives a much clearer instruction on how to resolve the error too